openpype/hosts/maya/plugins/publish/collect_pointcache.py

```diff
@@ -43,17 +43,34 @@ def process(self, instance):
             instance.remove(proxy_set)
             instance.data["setMembers"].remove(proxy_set)
 
-        # Collect user defined attributes.
-        if not instance.data.get("includeUserDefinedAttributes", False):
-            return
+        # Backwards compatibility for attributes.
+        backwards_mapping = {
+            "write_color_sets": "writeColorSets",
+            "write_face_sets": "writeFaceSets",
+            "include_user_defined_attributes": "includeUserDefinedAttributes"
+        }
+        for key, value in backwards_mapping.items():
+            if key in instance.data:
+                self.log.debug(
+                    "Using legacy attribute name '{}' since it exists.".format(
+                        key
+                    )
+                )
+                instance.data[value] = instance.data.pop(key)
 
-        user_defined_attributes = set()
-        for node in instance:
-            attrs = cmds.listAttr(node, userDefined=True) or list()
-            shapes = cmds.listRelatives(node, shapes=True) or list()
-            for shape in shapes:
-                attrs.extend(cmds.listAttr(shape, userDefined=True) or list())
+        # Collect user defined attributes.
+        if instance.data.get("includeUserDefinedAttributes", False):
+            user_defined_attributes = set()
+            for node in instance:
+                attrs = cmds.listAttr(node, userDefined=True) or list()
+                shapes = cmds.listRelatives(node, shapes=True) or list()
+                for shape in shapes:
+                    attrs.extend(
+                        cmds.listAttr(shape, userDefined=True) or list()
+                    )
 
-            user_defined_attributes.update(attrs)
+                user_defined_attributes.update(attrs)
 
-        instance.data["userDefinedAttributes"] = list(user_defined_attributes)
+            instance.data["userDefinedAttributes"] = list(
+                user_defined_attributes
+            )
```

Comment:
We should really look into ways to "deduplicate" this code amongst the two plugins. 
I'd actually go as far and recommend to put the logic for the backwards compatibility + the user defined attributes into plugins of their own that is set to the `families` of both of them.